documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,501 @@
+
+# ... don't worry about other files it creates
+
+Generating Migrations:
+* creating priv/repo/migrations/20230208045101_initial_migration.exs
+```
+
+Here is the migration file commented in detail:
+
+```elixir
+defmodule MyAshPhoenixApp.Repo.Migrations.InitialMigration do
+  use Ecto.Migration
+
+  # This function runs when migrating forward
+  def up do
+    # Creates the `:posts` table
+    create table(:posts, primary_key: false) do
+      # Adds primary key attribute `:id` of type `:uuid`
+      # null values are not allowed
+      add :id, :uuid, null: false, default: fragment("uuid_generate_v4()"), primary_key: true
+
+      # Adds attribute `:title` of type `:text`, null values are not allowed
+      add :title, :text, null: false
+      # Adds attribute `:content` of type `:text`, null values are allowed
+      add :content, :text
+    end
+  end
+
+  # This is the function that runs if you want to rollback the migration.
+  def down do
+    # Deletes the `:posts` table
+    drop table(:posts)
+  end
+end
+```
+
+We can run the `up/0` function which will perform the desired operations on the Postgres database. We do this with the migrate command:
+
+```bash
+$ mix ash_postgres.migrate
+```
+
+## Interacting with your Resources
+
+**All interaction with your resource attributes always occur through an action**. In our resource we are using the default actions for `:create, :read, :update, :destroy` along with a custom action `:by_id`.
+
+`:create` and `:update` and `:destroy` actions require a changeset. Ash changesets are conceptually similar to [Ecto changesets](https://hexdocs.pm/ecto/Ecto.Changeset.html). They're data structures which represent an intended change to an Ash resource and provide validation.
+
+The `:read` action takes a query instead of a changeset
+
+Below is the most verbose way of calling your resource. All other ways of interaction is some kind of shorthand of these. This means at some point a changeset is being created and passed to the API, even if its encapsulated within another function.
+
+```elixir
+# create post
+new_post =
+  MyAshPhoenixApp.Blog.Post
+  |> Ash.Changeset.for_create(:create, %{title: "hello world"})
+  |>  MyAshPhoenixApp.Blog.create!()
+
+# read all posts
+MyAshPhoenixApp.Blog.Post
+|> Ash.Query.for_read(:read)
+|> MyAshPhoenixApp.Blog.read!()
+
+# get single post by id
+MyAshPhoenixApp.Blog.Post
+|> Ash.Query.for_read(:by_id, %{id: first_post.id})
+|> MyAshPhoenixApp.Blog.read_one!()
+
+# update post
+updated_post =
+  new_post
+  |> Ash.Changeset.for_update(:update, %{content: "hello to you too!"})
+  |> MyAshPhoenixApp.Blog.update!()
+
+# delete post
+first_post
+|> Ash.Changeset.for_destroy(:destroy)
+|> MyAshPhoenixApp.Blog.destroy!()
+```
+
+As I said, this is verbose so Ash has a built in shortcut - The `code_interface`. You may notice this has already been done in your `Post` resource. Here it is again with more explanation:
```

Comment:
Great guide, this is exactly what I needed when I was starting out! Only a tiny comment, this looks like the only place you've first-person "As I said"

Reply:
https://github.com/chrishop/ash/blob/675a3a41730b743fd0bd8b5db419557a0068ff22/documentation/tutorials/get-started-with-ash-and-phoenix.md?plain=1#L317
I say before the code block "Below is the most verbose way of calling your resource."

Reply:
Oh no that's not quite what I meant sorry, I was referring to the voice that the guide is written in. But on re-reading, the style is conversational because you refer to "your" resources and "we/our" project so I think it makes sense, was just surprised the first time to hear "I". Nitpicking anyway 😅

Reply:
Might be worth rephrasing as `As stated above`

Reply:
Ahh ok I getya now! Yea that should change change it. The voice I speak in is to the user so I say 'your' quite a lot. But I think that is the only place I've referenced myself.

Reply:
addressed: [3f63e40](https://github.com/ash-project/ash/pull/504/commits/3f63e40a6bd0078c8a781b3fd8c8d7a6898e2e82)